packages/engine/Source/Scene/Cesium3DTilesetBaseTraversal.js

```diff
@@ -295,7 +295,9 @@ function executeEmptyTraversal(root, frameState) {
     }
   }
 
-  return allDescendantsLoaded;
+  return (
+    root.hasEmptyContent || root.hasImplicitContent || allDescendantsLoaded
+  );
```

Comment:
If a placeholder tile has an empty child we're going to see the same problems that came up in #9356.

To fix that I think another line should be changed:

```js
const traverse = (tile.hasTilesetContent || tile.hasImplicitContent) && canTraverse(tile);
```

In other words, stop traversal if an empty tile is hit, but continue if an external tileset or implicit subtree is hit.

Also, the `emptyLeaf` code can be removed. It's no longer necessary.

Reply:
I think there could be a cleaner way to do this and scrap `executeEmptyTraversal`, but I'm not exactly sure what that would look like.

Reply:
Maybe final version of the function? I made a couple other tweaks:

```js
function executeEmptyTraversal(root, frameState) {
  const {
    canTraverse,
    updateTile,
    loadTile,
    touchTile,
  } = Cesium3DTilesetTraversal;
  let allDescendantsLoaded = true;
  const stack = emptyTraversal.stack;
  stack.push(root);

  while (stack.length > 0) {
    emptyTraversal.stackMaximumLength = Math.max(
      emptyTraversal.stackMaximumLength,
      stack.length
    );

    const tile = stack.pop();
    const children = tile.children;
    const childrenLength = children.length;

    // Only traverse if the tile is an external tileset or implicit subtree
    const traverse = (tile.hasTilesetContent || tile.hasImplicitContent) && canTraverse(tile);

    // Traversal stops but the external tileset or implicit subtree hasn't loaded yet
    // There will be holes if the parent tries to refine to its children, so don't refine
    if (!traverse && !tile.contentReady) {
      allDescendantsLoaded = false;
    }

    updateTile(tile, frameState);
    if (!tile.isVisible) {
      // Load tiles that aren't visible since they are still needed for the parent to refine
      loadTile(tile, frameState);
      touchTile(tile, frameState);
    }

    if (traverse) {
      for (let i = 0; i < childrenLength; ++i) {
        const child = children[i];
        stack.push(child);
      }
    }
  }

  return root.hasEmptyContent || allDescendantsLoaded;
}
```

Reply:
@lilleyse I'm seeing holes in the Google tileset with this approach.

Reply:
I'm seeing the same thing as Gabby. The problem is the change to the initialization of `const traverse`.

I do think we can get rid of the `emptyLeaf` exception. It doesn't appear that is needed any longer according to my testing. Let me run the unit tests locally to confirm and I'll push up that change in a few moments.

Reply:
@ggetz @lilleyse When you two get a chance, please take a look at my latest push. It works pretty well in my testing and all of the tests pass with @ggetz's modifications.

Reply:
Sorry I haven't had a chance to get back to this.

I probably want have a chance to debug what's wrong with my code, so I'd be ok with proceeding with @weegeekps's changes. Let's just watch out for bugs where the root tile of an external tileset or implicit subtree is empty since I don't think that's being properly handled.

Reply:
Thanks @lilleyse! We're happy to review any further changes here. Let's proceed given @weegeekps's changes are a noticeable improvement on the existing behavior.